src/boot/efi/boot.c

```diff
@@ -43,7 +43,8 @@ enum loader_type {
         LOADER_SECURE_BOOT_KEYS,
 };
 
```

Comment:
so i take it this is edk2 specific? if so, please say so loudly in the commit msg

Reply:
Strictly speaking, no. Any firmware can bundle a UEFI shell in their firmware volume. They just tend not to?

Reply:
please add discussion of this to the commit msg, and as comment

Reply:
> Strictly speaking, no. Any firmware can bundle a UEFI shell in their firmware volume. They just tend not to?

I have in my hands hardware that has an UEFI shell built-in and exposed as a BootOption by default. So it does happen that real hardware ship an internal shell.